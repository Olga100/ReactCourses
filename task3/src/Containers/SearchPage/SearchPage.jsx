import React, { Component } from 'react';

import Header from '../Header/Header';
import MovieList from '../../Components/MovieList/MovieList';
import ErrorBoundary from '../../Components/ErrorBoundary';
import LineResultMovieList from '../../Components/LineResultMovieList/LineResultMovieList';
import getMovies from '../../api';

class SearchPage extends Component {
    constructor(props) {
        super(props);

        this.state = {
            movies: [],
            search: "",
            searchBy: "title",
            sortBy: "release_date"
        };
    }

    componentDidMount() {
        this.loadMovies();
    }

    loadMovies() {
        getMovies(this.state.search, this.state.searchBy, this.state.sortBy)
            .then(movies => this.setState({movies: movies.data }));
    }

    handleSortByChanged(sortBy) {
        this.setState(
            {sortBy: sortBy},
            () => this.loadMovies());
    }

    handleSearchByChanged(search, searchBy) {
        this.setState(
            {search: search, searchBy: searchBy},
            () => this.loadMovies());
    }

    render() {
        const {searchBy, sortBy, movies} = this.state;

        return (
            <div className="App">
                <ErrorBoundary>
                    <Header
                        searchBy={searchBy}
                        searchByChanged = {(search, searchBy) => this.handleSearchByChanged(search, searchBy)}
                    />

                    <LineResultMovieList
                        moviesCount={movies.length}
                        sortBy={sortBy}
                        sortByChanged={(sortBy) => {this.handleSortByChanged(sortBy)}}
                    />
                    <MovieList
                        movies={movies}
                    />
                </ErrorBoundary>
            </div>
        );
    }
}

export default SearchPage;
