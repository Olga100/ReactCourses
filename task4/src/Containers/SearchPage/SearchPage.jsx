import React, { Component } from 'react';

import Header from '../Header/Header';
import MovieList from '../../Components/MovieList/MovieList';
import ErrorBoundary from '../../Components/ErrorBoundary';
import LineResultMovieList from '../../Components/LineResultMovieList/LineResultMovieList';
import getMovies from '../../api';

class SearchPage extends Component {
    constructor(props) {
        super(props);

        this.state = {
            movies: [],
            search: "",
            searchBy: "title",
            sortBy: "release_date"
        };
    }

    componentDidMount() {
        this.loadMovies();
    }

    loadMovies() {
        const { search, searchBy, sortBy} = this.state;

        getMovies(search, searchBy, sortBy)
            .then(movies => this.setState({movies: movies.data }));
    };

    handleSortByChanged = (sortBy) => {
        this.setState({sortBy: sortBy},
            () => this.loadMovies());
    };

    handleSearchByChanged = (search, searchBy) => {
        this.setState({search: search, searchBy: searchBy},
            () => this.loadMovies());
    };

    render() {
        const {searchBy, sortBy, movies} = this.state;

        return (
            <div className="App">
                <ErrorBoundary>
                    <Header
                        searchBy={searchBy}
                        searchByChanged = {this.handleSearchByChanged}
                    />

                    <LineResultMovieList
                        moviesCount={movies.length}
                        sortBy={sortBy}
                        sortByChanged={this.handleSortByChanged}
                    />
                    <MovieList
                        movies={movies}
                    />
                </ErrorBoundary>
            </div>
        );
    }
}

export default SearchPage;
