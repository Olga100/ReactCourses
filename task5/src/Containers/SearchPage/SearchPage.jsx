import React, { Component } from 'react';
import {connect} from 'react-redux';

import Header from '../../Components/Header/Header';
import MovieList from '../../Components/MovieList/MovieList';
import ErrorBoundary from '../../Components/ErrorBoundary';
import LineResultMovieList from '../../Components/LineResultMovieList/LineResultMovieList';
//import getMovies from '../../api';
import {loadMovies, sortBy, searchBy, searchText} from '../../Actions/Actions';

class SearchPageView extends Component {
    constructor(props) {
        super(props);

    }

    componentDidMount() {
        this.props.loadMovies();
    }


    handleSortByChanged = (sortBy) => {
        this.props.sortByChanged(sortBy);
        this.props.loadMovies(this.props.searchText, this.props.searchBy, sortBy);
    };


    handleSearch = () => {
        this.props.loadMovies(this.props.searchText, this.props.searchBy, this.props.sortBy);
    };

    render() {
        const { movies, sortBy, searchBy, searchText, searchByChanged, searchTextChanged }  = this.props;

        return (
            <div className="App">
                <ErrorBoundary>
                    <Header
                        searchText={searchText}
                        searchBy={searchBy}
                        searchTextChanged={searchTextChanged}
                        searchByChanged={searchByChanged}
                        search={this.handleSearch}
                    />

                    <LineResultMovieList
                        moviesCount={movies.length}
                        sortBy={sortBy}
                        sortByChanged={this.handleSortByChanged}
                    />
                    <MovieList
                        movies={movies}
                    />
                </ErrorBoundary>
            </div>
        );
    }
}


const mapStateToProps =  (state) => ({
    movies: state.viewState.movies,
    sortBy: state.viewState.sortBy,
    searchBy: state.viewState.searchBy,
    searchText: state.viewState.searchText
});

const mapDispatchToProps = (dispatch) => ({
    loadMovies: (search, searchBy, sortBy) => dispatch(loadMovies(search, searchBy, sortBy)),
    sortByChanged: (x) => dispatch(sortBy(x)),
    searchByChanged: (x) => dispatch(searchBy(x)),
    searchTextChanged: (text) => dispatch(searchText(text))
});

const SearchPage = connect(mapStateToProps, mapDispatchToProps)(SearchPageView);

export default SearchPage;
