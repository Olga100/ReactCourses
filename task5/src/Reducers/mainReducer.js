import * as actions from '../Actions/Actions';

export function mainReducer(state = {}, action) {

    if(action.type === actions.RECEIVE_MOVIES) {
        const newState = Object.assign({}, state);
        newState.viewState.movies = action.movies;
        return newState;
    }

    if(action.type === actions.SORT_BY) {
        const newState = Object.assign({}, state);
        newState.viewState.sortBy = action.field;
        return newState;
    }

    if(action.type === actions.SEARCH_BY) {
        const newState = Object.assign({}, state);
        newState.viewState.searchBy = action.field;
        return newState;
    }

    if(action.type === actions.SEARCH_TEXT) {
        const newState = Object.assign({}, state);
        newState.viewState.searchText = action.text;
        return newState;
    }

    return state;
}
