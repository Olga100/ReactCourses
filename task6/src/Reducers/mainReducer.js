import {RECEIVE_MOVIES, SORT_BY, SEARCH_BY, SEARCH_TEXT,  SELECTED_MOVIE} from'../constants';

const initialState = {
    movies: [],
    sortBy: 'release_date',
    searchBy: 'title',
    searchText: "",
    movie: {}
};

export default function mainReducer(state = initialState, action) {
    const { type } = action;
    console.log(action)

    switch (type) {
        case RECEIVE_MOVIES:
            return {...state, movies: action.movies};
        case SORT_BY:
            return {...state, sortBy: action.field};
        case SEARCH_BY:
            return {...state, searchBy: action.field};
        case SEARCH_TEXT:
            return {...state, searchText: action.text};
        case SELECTED_MOVIE:
            return {...state, movie: action.movie};
        default:
            return state;
    }
}

export const getQuery = state => {
    let {searchText, searchBy, sortBy} = state;
    let params = [];

    if (searchText) {
        params.push(`search=${searchText}`);
    }
    if (searchBy) {
        params.push(`searchBy=${searchBy}`);
    }
    if (sortBy) {
        params.push(`sortBy=${sortBy}`);
        params.push('sortOrder=desc');
    }
    return 'https://reactjs-cdp.herokuapp.com/movies?' + params.join('&');
};
