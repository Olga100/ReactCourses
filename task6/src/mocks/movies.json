{
  "movies": [
    {
      "id": 1,
      "title": "Star war",
      "poster_path": "https:\/\/images-na.ssl-images-amazon.com\/images\/M\/MV5BODU4MjU4NjIwNl5BMl5BanBnXkFtZTgwMDU2MjEyMDE@._V1_SY500_CR0,0,336,500_AL_.jpg",
      "genres": [
        "Crime",
        "Drama"
      ],
      "release_date": "2011-10-05"
    },
    {
      "id": 2,
      "title": "Star wars",
      "poster_path": "https:\/\/images-na.ssl-images-amazon.com\/images\/M\/MV5BZTRmNjQ1ZDYtNDgzMy00OGE0LWE4N2YtNTkzNWQ5ZDhlNGJmL2ltYWdlL2ltYWdlXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SY500_CR0,0,352,500_AL_.jpg",
      "genres": [
        "Crime",
        "Drama"
      ],
      "release_date": "2012-10-05"
    }
  ]
}