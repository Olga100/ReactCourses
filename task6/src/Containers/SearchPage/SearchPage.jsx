import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import Header from '../../Components/Header/Header';
import MovieList from '../../Components/MovieList/MovieList';
import ErrorBoundary from '../../Components/ErrorBoundary';
import LineResultMovieList from '../../Components/LineResultMovieList/LineResultMovieList';
import { loadMovies, sortBy, searchBy, searchText } from '../../Actions/Actions';
import { getQuery } from '../../Reducers/mainReducer';


class SearchPageView extends Component {
    componentDidUpdate(prevProps) {
        if (this.props.location !== prevProps.location) {
            const searchText = this.props.match.params.text;

            if (searchText) {
                const {searchTextChanged, loadMovies} = this.props;

                searchTextChanged(searchText);
                setTimeout(() => loadMovies(this.props.query));
            }
        }
    }

    handleSortByChanged = (sortBy) => {
        const { sortByChanged } = this.props;

        sortByChanged(sortBy);
        setTimeout(() => this.handleSearch());
    };


    handleSearch = () => {
        this.props.history.push('/search/' + encodeURI(this.props.searchText));
    };

    render() {
        const { movies, sortBy, searchBy, searchText, searchByChanged, searchTextChanged }  = this.props;

        return (
            <div className="App">
                <ErrorBoundary>
                    <Header
                        searchText={searchText}
                        searchBy={searchBy}
                        searchTextChanged={searchTextChanged}
                        searchByChanged={searchByChanged}
                        search={this.handleSearch}
                    />

                    <LineResultMovieList
                        moviesCount={movies.length}
                        sortBy={sortBy}
                        sortByChanged={this.handleSortByChanged}
                    />
                    <MovieList
                        movies={movies}
                    />
                </ErrorBoundary>
            </div>
        );
    }
}


const mapStateToProps =  (state) => ({
    movies: state.movies,
    sortBy: state.sortBy,
    searchBy: state.searchBy,
    searchText: state.searchText,
    query: getQuery(state)
});

const mapDispatchToProps = {
    loadMovies,
    sortByChanged: sortBy,
    searchByChanged: searchBy,
    searchTextChanged: searchText
};

const SearchPage = connect(mapStateToProps, mapDispatchToProps)(withRouter(SearchPageView));

export default SearchPage;
