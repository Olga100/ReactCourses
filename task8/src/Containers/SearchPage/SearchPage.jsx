import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import Header from '../../Components/Header/Header';
import MovieList from '../../Components/MovieList/MovieList';
import ErrorBoundary from '../../Components/ErrorBoundary';
import LineResultMovieList from '../../Components/LineResultMovieList/LineResultMovieList';
import { loadMovies, sortBy, searchBy, searchText } from '../../Actions/Actions';
import { getQuery } from '../../Reducers/mainReducer';

class SearchPageView extends Component {
    componentDidMount() {
        this.refreshSearch();
    }

    componentDidUpdate(prevProps) {
        if (this.props.location !== prevProps.location) {
            this.refreshSearch();
        }
    }

    handleSortByChanged = (sortBy) => {
        const { sortByChanged } = this.props;

        sortByChanged(sortBy);
        setTimeout(() => this.handleSearch());
    };


    handleSearch = () => {
        this.props.history.push('/search?' + this.props.query);
    };

    refreshSearch = () => {
        const {searchTextChanged, searchByChanged, sortByChanged, loadMovies} = this.props;
        const params = new URLSearchParams(this.props.location.search);

        searchTextChanged(params.get("search"));
        searchByChanged(params.get("searchBy"));
        sortByChanged(params.get("sortBy"));

        setTimeout(() => loadMovies(this.props.query));
    };

    render() {
        const { movies, sortBy, searchBy, searchText, searchByChanged, searchTextChanged }  = this.props;

        return (
            <div className="App">
                <ErrorBoundary>
                    <Header
                        searchText={searchText}
                        searchBy={searchBy}
                        searchTextChanged={searchTextChanged}
                        searchByChanged={searchByChanged}
                        search={this.handleSearch}
                    />

                    <LineResultMovieList
                        moviesCount={movies.length}
                        sortBy={sortBy}
                        sortByChanged={this.handleSortByChanged}
                    />
                    <MovieList
                        movies={movies}
                    />
                </ErrorBoundary>
            </div>
        );
    }
}


const mapStateToProps =  (state) => ({
    movies: state.get("movies"),
    sortBy: state.get("sortBy"),
    searchBy: state.get("searchBy"),
    searchText: state.get("searchText"),
    query: getQuery(state.toJS())
});

const mapDispatchToProps = {
    loadMovies,
    sortByChanged: sortBy,
    searchByChanged: searchBy,
    searchTextChanged: searchText
};

const SearchPage = withRouter(connect(mapStateToProps, mapDispatchToProps)(SearchPageView));

export default SearchPage;
