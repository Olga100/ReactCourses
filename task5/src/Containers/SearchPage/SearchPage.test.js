import mainReducer from '../../Reducers/mainReducer';
import thunkMiddleware from 'redux-thunk';
import expect from 'expect';
import SearchPage from './SearchPage';

import movies  from '../../mocks/movies.json';

const initialState = {
    movies: [],
    sortBy: 'release_date',
    searchBy: 'title',
    searchText: ""
};

describe ('SearchPage', () => {

    it('should match snapshot with data', () => {

        const mock = new MockAdapter(axios);
        mock.onGet('https://reactjs-cdp.herokuapp.com/movies?').reply(200, {data: []});
        mock.onGet('https://reactjs-cdp.herokuapp.com/movies?searchBy=title&sortBy=release_date&sortOrder=desc').reply(200, {data: []});

        const state = {
            movies: [],
            sortBy: 'release_date',
            searchBy: 'title',
            searchText: ""
        };

        const store = createStore(mainReducer, initialState, applyMiddleware(thunkMiddleware));

        const component = renderer.create(
            <Provider store={store}>
                <SearchPage />
            </Provider>
        );
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it('should handle store "movies" changes', () => {
        const searchText = "Star war";

        const expectedMovies =
            [
                {
                    id: 1,
                    title: "Star war",
                    poster_path: "http://www.theamazingpics.com/wp-content/uploads/2014/05/Amazing-Picture-of-A-Japanese-Garden-in-Portland-USA.jpg",
                    genres: ["horror"],
                    release_date: "2011-10-05T14:48:00.000Z"
                }
            ];

        const expectedStore = {
            movies: expectedMovies,
            sortBy: 'release_date',
            searchBy: 'title',
            searchText: searchText
        };

        const mock = new MockAdapter(axios);
        mock.onGet('https://reactjs-cdp.herokuapp.com/movies?').reply(200, {data: []});
        mock.onGet('https://reactjs-cdp.herokuapp.com/movies?searchBy=title&sortBy=release_date&sortOrder=desc').reply(200, {data: []});
        mock.onGet(`https://reactjs-cdp.herokuapp.com/movies?search=${searchText}&searchBy=title&sortBy=release_date&sortOrder=desc`).reply(200, {data: expectedMovies});

        const store = createStore(mainReducer, initialState, applyMiddleware(thunkMiddleware));

        const content = mount(
            <Provider store={store}>
                <SearchPage />
            </Provider>
        );

        content.find('input.searchMovie').simulate('change', {target: {value: searchText}});
        const form = content.find('form').at(0);
        form.simulate('submit');

        setImmediate(() => {
            expect(store.getState()).toEqual(expectedStore);
        });
    });
});
