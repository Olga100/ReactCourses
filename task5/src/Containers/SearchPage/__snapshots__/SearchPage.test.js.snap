// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should match snapshot with data 1`] = `
<div
  className="App"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="header"
    >
      <span
        className="netflex"
      >
        netflexroulette
      </span>
      <span
        className="titleFind"
      >
        FIND YOUR MOVIE
      </span>
      <div
        className="inputContainer"
      >
        <input
          className="searchMovie"
          onChange={[Function]}
          placeholder="search your movie"
          value=""
        />
      </div>
      <div
        className="filterContainer"
      >
        <div
          className="filterOptionsContainer"
        >
          <span
            className="searchTitle"
          >
            SEARCH BY
          </span>
          <button
            className="activeButton"
            id="title"
            onClick={[Function]}
            type="button"
          >
            TITLE
          </button>
          <button
            className={null}
            id="genre"
            onClick={[Function]}
            type="button"
          >
            GENRE
          </button>
        </div>
        <button
          className="search"
        >
          SEARCH
        </button>
      </div>
    </div>
  </form>
  <div
    className="lineResultContainer"
  >
    <span>
      0
       movies found
    </span>
    <span
      className="sortByWrapper"
    >
      <span>
        Sort by
      </span>
      <span
        className="active"
        id="sortByRelease"
        onClick={[Function]}
      >
        release date
      </span>
      <span
        className={null}
        id="sortByRating"
        onClick={[Function]}
      >
        rating
      </span>
    </span>
  </div>
  <div
    className="mainContainer"
  >
    <div
      className="movieListContainer"
    >
      <div
        className="noFilms"
      >
        No films found
      </div>
    </div>
  </div>
</div>
`;
